Sources/ClientGtk/ServerListView.swift

```diff
@@ -0,0 +1,93 @@
+import SwiftCrossUI
+import DeltaCore
+
+enum DetailState {
+  case addition, server(ServerDescriptor), error(String)
+}
+
+class ServerListViewState: Observable {
+  @Observed var servers: [ServerDescriptor] = []
+  @Observed var detailState = DetailState.addition
+}
+
+struct ServerListView: View {
+  var completionHandler: (ServerDescriptor) -> Void
+
+  var state = ServerListViewState()
+
+  var body: some ViewContent {
+    NavigationSplitView {
+      VStack {
+        Button("Add server") {
+          state.detailState = .addition
+        }
+
+        ScrollView {
+          ForEach(state.servers) { server in
+            Button(server.name) {
+              state.detailState = .server(server)
+            }
+            .padding(Edge.Set.bottom, 5)
+          }
+        }
+      }
+      .frame(minWidth: 100)
+      .padding(Edge.Set.trailing, 10)
+    } detail: {
+      VStack {
+        switch state.detailState {
+          case .addition:
+            additionView
+          case .server(let server):
+            Text(server.name)
+            if let port = server.port {
+              Text(server.host + ":" + String(port))
+            } else {
+              Text(server.host)
+            }
+
+            Button("Connect") {
+              completionHandler(server)
+            }
+          case .error(let message):
+            Text(message)
+        }
+      }
+      .padding(Edge.Set.leading, 10)
+    }
+  }
+
+  var additionView: some View {
+    VStack {
+      @Observed var name = ""
+      @Observed var address = ""
+
+      Text("Add server")
+      TextField("Name", $name)
+      TextField("Address", $address)
+      
+      Button("Add") {
+        let parts = address.split(separator: ":")
```

Comment:
To work around the crash for now, just introduce an extra check above this,

```swift
guard address != "", address != ":" else {
  state.detailState = .error("Empty host")
  return
}
```

I think that should cover the two edge cases that crash `split`.

Reply:
No; “::::”, “::” etc all crash it too - I think it would be best to leave this as-is and wait for the bug to be resolved.

Reply:
Ah ok, yeah, that is a very weird crash lol